src/include/duckdb/optimizer/rule/join_dependent_filter.hpp

```diff
@@ -0,0 +1,24 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/optimizer/rule/join_filter_derivation.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/optimizer/rule.hpp"
+
+namespace duckdb {
+
+//! Join-dependent filters cannot be pushed down, but we can sometimes derive a filter from them that can
```

Comment:
Could we expand the description here to contain the description of the PR? It is not very clear what this rule does by itself.